src/main/java/core/basesyntax/strategy/impl/BalanceOperationHandler.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.transaction.FruitTransaction;
+
+public class BalanceOperationHandler implements OperationHandler {
+    @Override
+    public void doCalculation(FruitTransaction fruitTransaction) {
+        String fruit = fruitTransaction.getFruit();
+        int quantity = fruitTransaction.getQuantity();
+        if (Storage.storage.containsKey(fruit)) {
+            int newQuantity = Storage.storage.get(fruit)
+                    + quantity;
+            Storage.storage.put(fruit, newQuantity);
+        } else {
+            Storage.storage.put(fruit, quantity);
+        }
```

Comment:
Consider using `Map.getOrDefault`.

```suggestion
            int newQuantity = Storage.storage.getOrDefault(fruit, 0) + quantity;
            Storage.storage.put(fruit, newQuantity);
```